docs/CODE_GUIDELINES.md

```diff
@@ -280,82 +415,200 @@ enum Dummy
 };
 ```
 
-### 11.4. Interfaces
-Use CamelCase for interface names and they have to be prefixed with an uppercase I. Filename has to match the interface name without the prefixed I, e.g. Logger.h
+### 8.4. Interfaces
+Use PascalCase and prefix with an uppercase I. Filename has to match the interface name without the prefixed I, e.g. Logger.h
 ```cpp
 class ILogger
 {
-  void Log(...) = 0;
+public:
+  virtual void Log(...) = 0;
+  virtual ~ILogger() {}
 }
 ```
 
-### 11.5. Classes
-Use CamelCase for class names and they have to be prefixed with an uppercase C. Filename has to match the class name without the prefixed C, e.g. Logger.cpp
+### 8.5. Classes
+Use PascalCase and prefix with an uppercase C. Filename has to match the class name without the prefixed C, e.g. Logger.cpp
 ```cpp
 class CLogger : public ILogger
 {
-  void Log(...)
+public:
+  virtual void Log(...) override;
 }
 ```
 
-### 11.6. Methods
-Use CamelCase for method names and first letter has to be uppercase, even if the methods are private or protected.
+### 8.6. Methods
+Use PascalCase always, uppercasing the first letter even if the methods are private or protected.
 ```cpp
 void MyDummyClass::DoSomething();
 ```
 
-### 11.7. Variables
-Use CamelCase for variables. Type prefixing is discouraged.
+### 8.7. Variables
+Use CamelCase. Type prefixing (Systems Hungarian notation) is discouraged.
+
+#### Member variables
+Prefix nonstatic member variables with `m_`. Prefix static member variables with `ms_`.
+```cpp
+int m_variableA;
+static int ms_variableB;
```

Comment:
We don't use this, but I like it.

Reply:
Oh snap, you're right. Must have mixed it up with some other project. Anyway I can leave it in if people think it helps.